Jenkinsfile

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env groovy
+pipeline {
+  agent {
+    dockerfile true
+  }
+
+  environment {
+    JENKINS_VERSION = 'yes'
+    NODE_PATH = 'node_modules'
+  }
+
+  options {
+    timeout(time: 2, unit: 'HOURS')
+  }
+
+  stages {
+
+    stage('Initialization') {
+      steps {
+        ansiColor('xterm') {
+          echo 'Setting-up environment...'
+
+          // Install might fail with 'unexpected eof'
+          retry(2) {
+            sh '''npm --unsafe-perm install'''
+          }
+
+          // Install npm dependencies
+          sh '''npm run scaffold'''
+        }
+      }
+    }
+
+    stage('Lint') {
+      steps {
+        ansiColor('xterm') {
+          sh '''npm run lint'''
+        }
+      }
+    }
+
+    stage('Unit Tests') {
+      steps {
+        ansiColor('xterm') {
+          sh '''npm run test'''
+        }
+
+        junit 'jest/test-results/*.xml'
+      }
+    }
+
+    stage('Build') {
+      steps {
+        ansiColor('xterm') {
+          sh '''npm run build-assets'''
+        }
+        stash includes: 'dist/*', name: 'dist'
+      }
+    }
+
+    stage('Integration Tests') {
```

Comment:
I believe we would benefit a lot in performance if we use the `parallel` feat from jenkins pipeline.
The places I believe we can use is `Lint` `Unit test` `Integration tests` and `System test`

What you think @orlandohohmeier 

for more info: https://www.cloudbees.com/blog/parallelism-and-distributed-builds-jenkins

Reply:
@weblancaster I had a version that used parallel steps, but ran into memory issues as all of these test are executed on the same machine if we don't use the `node` directive to schedule the work on a different nodes and I haven't figured out how to share the initially build docker image which results in rather long bootstrap times and thus is not faster than running these synchronously. Anyway, I'll continue investigating ways to improve the performance and build time and really appreciate your comments and ideas.

Reply:
Ok cool, thanks for the feedback.